### ISSUE TITLE
Add option for collection managers to enter keywords at the collection level that will be applied to all entries for a collection.

### ISSUE THREAD

**Issue by amyehodge** [sul-dlss/happy-heron#1738]
This will require 
- an addition to the collection form for entering these keywords. This should function like the one on the deposit form, with FAST lookup and free text entry
- a display on the collection view page of keywords entered
- a display on the item deposit page of keywords to be entered that are not changeable
- change to the item view page showing these additional terms. these could be listed separately from those defined by the user, or together.

@astridu Could you mock some things up for this one?

We should complete the work in #1736 and #1737 before addressing this ticket.

**Comment by amyehodge** [comment]
Collection edit page mockup: https://projects.invisionapp.com/share/EQXC9CLKCR2#/screens/454558426_keyword_In_Collection_Edit

Collection view page mockup: https://projects.invisionapp.com/share/EQXC9CLKCR2#/screens/454558535_keyword_In_Collection_View

Item deposit page mockup: 

Item view page mockup:

**Comment by astridu** [comment]
@amyehodge What do you think of this? Item view page mockup:
![image](https://user-images.githubusercontent.com/42216982/125365447-1dea7100-e329-11eb-806b-3ca7803cd0a9.png)

**Comment by amyehodge** [comment]
@astridu This looks good. I like having them in separate sections like this. I'm iffy on the term "collection designated" and thinking about something like "preassigned." What do you think?

**Comment by astridu** [comment]
That works!
![image](https://user-images.githubusercontent.com/42216982/125550268-1124e2c3-5777-4eae-b831-aa38450e669d.png)

**Comment by jcoyne** [comment]
We don't have a way on the item of differentiating between what the user entered and what was a collection default.  Those all end up in the same place in the metadata.  Is that an important distinction to make?

**Comment by amyehodge** [comment]
Ah. I think it's nice for the user to be able to see this on the item view, but not critical if it's complicated. However, depositors *should* be able to see the pre-assigned collection level terms on the deposit for as Astrid has mocked it up so they 1) know these terms will be assigned and 2) won't enter them again and 3) won't be able to alter or delete them like they would if they just showed up as normal entries. That may mean we need a way to differentiate these in the metadata, which would allow us to list them separately here.

**Comment by jcoyne** [comment]
@amyehodge what if they make version 1 and "topic 1" is required, but when the come back to edit, the collection has been changed and "topic 1" is not required but "topic 2" is now required?  How would we like it to display in those cases?

**Comment by amyehodge** [comment]
@jcoyne We should change the keywords assigned to the item at the collection level and display those. So:

1. Collection manager assigns one or more keywords at the collection level.
2. Depositor creates an item and those terms are assigned. I think let's still require them to enter at least one keyword of their own here and not count the terms set at the collection level.
3. Depositor completes deposit.
4. Collection manager changes the collection-level terms.
5. Depositor returns to version the item after collection-level terms have changed.
6. During versioning we should update the collection-level terms to the new ones. The depositor should just see the new terms displayed. I don't think it's necessary to let them know what they were or that they have changed.
7. Depositor completes deposit of item, which will now have updated collection-level keywords.

**Comment by jcoyne** [comment]
@amyehodge I'm concerned that we could surprise the depositor if we remove keywords that they thought were already associated with their deposit.

**Comment by amyehodge** [comment]
@jcoyne That is a concern. The collection manager might remove a term that the user wants to have present. The user wouldn't have added it originally because it was already there, but they might not notice that it has been removed. So, I guess we could alert them when a term has been removed? I'm not sure what that would look like.

This also makes me think that we should make sure it's clear that terms added will only be applied to new deposits or new versions of deposits and not to existing items.

@astridu Can you help with these things? For this second point, I think we may just be able to change the text from "If you would like all deposits to this collection to have certain keywords, enter them below. If not, leave blank" to something like "Enter any keywords you would like applied to all new deposits or new versions of existing deposits in this collection. Existing items will not be updated." Other suggestions for verbiage welcome.

**Comment by amyehodge** [comment]
@jcoyne, I discussed this some with @astridu this morning regarding showing the changed collection-level terms to depositors. I want to run through how this would work.

When a collection manager sets keywords, they will only be assigned to new deposits or newly versioned items in a collection, not to any existing items. They will have to go back and change those by hand at the moment if they want that to happen. I think this is the way to proceed right now, because otherwise we'd have to have a way to version all the existing objects when the manager sets this. That sounds very complicated.

So, let's say the collection manager adds or removes a collection-level keyword after an item has been deposited, and then the depositor logs in to version that deposit. My suggestion was that we just display the current collection-level keywords and update them when the item is deposited. If we want let the user know what has been added or removed, then we are going to have track the dates when each term was added or removed or modified and then check that against the date the item was opened back up again to know which ones have changed since that item was deposited. If the depositor keeps the new version open for a long time, it's also possible the terms will change again, so we'll have to check and update that display every time it's opened until it gets deposited again. 

I'm wondering how much complexity this adds. If that's all easily doable, then we can have @astridu mock up how to handle displaying the changes to the user.

**Comment by astridu** [comment]
Amy and I discussed, and we decided on the following:
1. update text on collection details page where user enters keywords for collections to "Enter keywords you would like applied to new deposits or to new versions of existing deposits in this collection. Existing items will not be updated."
2. Indicate on the view item page that a new keyword has been added to the preassigned with a tooltip for the field label in which we could explain that the collection manager may have added/deleted keywords to the collection with the latest version update -- or something to that effect. 
![image](https://user-images.githubusercontent.com/42216982/125867134-f40d7e8c-dc3a-4748-8790-2eef06e80281.png)
![image](https://user-images.githubusercontent.com/42216982/125867186-f83ffe3f-0a05-43ba-9c1a-b6736af7c921.png)
